core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -14,14 +15,31 @@ import scala.annotation.implicitNotFound
   * Chisel.
   */
 @implicitNotFound("unsupported Property type ${T}")
-private[chisel3] class PropertyType[T]
+private[chisel3] trait PropertyType[T] {
+
+  /** Get the IR PropertyType for this PropertyType.
+    */
+  def getPropertyType(): ir.PropertyType
+}
 
 /** Companion object for PropertyType.
   *
   * Typeclass instances for valid Property types are defined here, so they will
   * be in the implicit scope and available for users.
   */
-private[chisel3] object PropertyType {}
+private[chisel3] object PropertyType {
+  implicit val intPropertyTypeInstance = new PropertyType[Int] {
+    def getPropertyType(): ir.PropertyType = ir.IntegerPropertyType
```

Comment:
Nit: it's best practice to use the `override` keyword whenever you override an existing method. Scala 2 doesn't require this for whatever reason.

Reply:
ah yes, makes sense